webrtc/RTCPeerConnection-iceConnectionState.https.html

```diff
@@ -371,4 +371,29 @@
   assert_array_equals(pc2.iceStates, ['new', 'checking', 'connected']);
 }, 'Responder ICE connection state behaves as expected');
 
+/*
+  Test case for step 11 of PeerConnection.close().
+  ...
+  11. Set connection's ICE connection state to "closed". This does not invoke
+      the "update the ICE connection state" procedure, and does not fire any
+      event.
+  ...
+*/
+promise_test(async t => {
+  const pc1 = new RTCPeerConnection();
+  t.add_cleanup(() => pc1.close());
+  const pc2 = new RTCPeerConnection();
+  const stream = await getNoiseStream({ audio: true });
+  t.add_cleanup(() => stream.getTracks().forEach(track => track.stop()));
+
+  stream.getTracks().forEach(track => pc1.addTrack(track, stream));
+  exchangeIceCandidates(pc1, pc2);
+  doSignalingHandshake(pc1, pc2);
+  await listenToIceConnected(pc2);
+
+  pc2.oniceconnectionstatechange = t.unreached_func();
+  pc2.close();
```

Comment:
Optional suggestion: Add this line before asserting:
await Promise.resolve();

This might catch an implementation that surfaces the invalid iceConnectionState change in a posted task

Reply:
Good point, or the test may end prematurely. Probably even a t.set_timeout or whatever it is called.

Reply:
Thanks for your suggestion. A new patch was submitted. Please help merge this PR if it looks good to you. I don't have write access to merge it. Thanks.